text/0116-sentry-semantic-conventions.md

```diff
@@ -0,0 +1,159 @@
+
+The reason for having an independent set of semantic conventions is that we have some attributes that are not covered by OpenTelemetry's semantic conventions, and we don't want to be blocked on upstreaming them. There are also some Sentry specific attributes that will never match OpenTelemetry's semantic conventions.
+
+For sentry specific fields like `release` and `environment`, we are proposing setting them under the `sentry.X` namespace for the key. For example, `release` would be defined under the `sentry.release` attributes key.
+
+## Attributes Schema Versioning and Implementation
+
+The attributes schema will be versioned and published as a JSON Schema, which will be the representation of the well-known attributes that Sentry ingestion and product relies on.
+
+The JSON schema will also be codegened into packages for JavaScript, Python, and Rust, so that the schema can be used in Relay, the frontend and the backend. There has to be care put in to make sure that the codegened packages are compatible with the JSON schema, and that the JSON schema is backwards compatible, but we can introduce tests to make sure that this is the case.
+
+By having a versionig/packaging structure like this, it also makes it much easier for the SDKs to identify what parts of the product are using the data they are sending.
+
+## Mapping Sentry Specific Fields to Attributes
+
+Right now we have a variety of top level fields that need to be mapped to attributes, and some of them match existing OpenTelemetry semantic conventions. We should map these fields to attributes as follows (the dots in the initial fields represent nested fields):
+
+Top level fields:
+
+- `release` -> `sentry.release`
+- `environment` -> `sentry.environment`
+
+[User interface](https://develop.sentry.dev/sdk/event-payloads/user/):
+
+- `user.id` -> `sentry.user.id`
+- `user.username` -> `sentry.user.username`
+- `user.email` -> `sentry.user.email`
+- `user.ip_address` -> `sentry.user.ip_address`
+- `user.segment` -> `sentry.user.segment`
+- `user.geo.city` -> `sentry.user.geo.city`
+- `user.geo.country_code` -> `sentry.user.geo.country_code`
+- `user.geo.region` -> `sentry.user.geo.region`
+
+[Request interface](https://develop.sentry.dev/sdk/event-payloads/request/)
+
+- `request.method` -> [`http.request.method`](https://github.com/open-telemetry/semantic-conventions/blob/cadfe53949266d33476b15ca52c92f682600a29c/model/trace/http.yaml#L19)
+- `request.url` -> [`url.full`](https://github.com/open-telemetry/semantic-conventions/blob/cadfe53949266d33476b15ca52c92f682600a29c/model/trace/http.yaml#L47)
+- `request.query_string` -> [`url.query_string`](https://github.com/open-telemetry/semantic-conventions/blob/cadfe53949266d33476b15ca52c92f682600a29c/model/trace/http.yaml#L47)
+- `request.cookies` -> `http.request.cookies`
+- `request.cookies` -> `http.request.headers`
```

Comment:
```suggestion
- `request.headers` -> `http.request.headers`
```

Is the idea of cookies / headers to store a JSON string or to store them in nested dot notation, e.g.

```js
span.setAttribute('request.headers.content-type', 'application/json');
```

Reply:
I'm still undecided if we should do 

> we will allow dictionaries as attribute values for better backwards compatibility with existing Sentry fields

The idea here was to use a JSON string because it is the easiest to adapt from what we have, but maybe we should strictly enforce the otel attributes schema of no nesting. Still going back and forth on it.